src/main/java/org/hypertrace/gradle/ci/tasks/CopyReportsTask.java

```diff
@@ -37,33 +34,30 @@ public File getNamespacedOutputDir() {
 
   public CopyReportsTask() {
     this.setOutputDir(new File(getProject().getBuildDir(), "collected-reports"));
+    this.dependsOn(this.getReportingTasks(getProject()));
   }
 
   @TaskAction
   public void copyFiles() {
-    this.getReportsForProject(this.getProject())
+    this.getReportingTasks(this.getProject())
+        .getAsMap()
         .forEach(
-            report -> {
-              File targetDirectoryForReport =
-                  new File(this.getNamespacedOutputDir(), report.getName());
+            (reportTaskName, reportTask) -> {
+              File targetDirectoryForReportTask =
+                  new File(this.getNamespacedOutputDir(), reportTaskName);
               this.getProject()
                   .copy(
                       copySpec -> {
-                        copySpec.from(report.getOutputLocation());
```

Comment:
The return type here changed from provider to property. Effectively, compatibility is the same but this means that newer versions of gradle won't work with the old one, and if I recompile with new gradle it's a breaking change.

https://docs.gradle.org/current/userguide/upgrading_version_7.html#report_getoutputlocation_return_type_changed_from_provider_to_property

Instead, what I did was to collect the reporting tasks (that contain reports) instead of the actual reports themselves, and copy each reporting task's outputs rather than each report's outputs. Effectively, the same thing (other than some potential small changes in output tree structure, but since this is used for archiving not a big deal) - but this way it's compatible with both existing and new gradle versions.